appsmithorg/appsmith
# feat: Table widget checkbox column type

## Description

This PR introduces a new column type: `Checkbox`. It uses checkbox widget's component to render the checkbox inside the table's cell. 

Here is the list of changes that are implemented:
* New column type added: Checkbox.
* Property pane level options:
     * Computed Value
     * Visible
     * Editable
     * Styles:
          * Horizontal Alignment
          * Vertical Alignment
          * Cell background
     * Events:
          * onCheckChange: This is available only when editable is true
* Table can now be filtered with is Checked and is unChecked conditions

**_NOTE_**: When the Editable property is false then the checkboxes are disabled.

Please refer to the following for more information:
* [PRD](https://www.notion.so/appsmith/Checkbox-in-table-687ad8bcf19243428c17ad5f7af9e0c4)
* [Figma](https://www.figma.com/file/aJ2Hwz9gH5AK6zaRYtZV8a/Table-widget-update?node-id=527%3A46002)

Fixes #7338 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

* Cypress spec: `checkboxCell_spec.js` is added. It covers the following scenarios:
    * Checks if the checkbox column type is available.
    * Checks if the computed value is editable and `currentRow` object is available.
    * Verifies visibility of the column
    * When editable is `true` then check:
         * If check box is enabled 
         * If `onCheckChange` is available
         * If handler attached to `onCheckChange` is working via interacting with the checkbox
    * if editable is `false` then verify if checkboxes are disabled
    * Check horizontal, vertical alignment and cell background color
    * Check if the filter conditions: `is checked` and `is unchecked` are working as expected.
    * Verify that `onCheckChange` is not available on `custom columns`

## Checklist:
